pkg/client/redhatsso/client_test.go

```diff
@@ -0,0 +1,696 @@
+package redhatsso
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/keycloak"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/test/mocks"
+	. "github.com/onsi/gomega"
+	"github.com/patrickmn/go-cache"
+	serviceaccountsclient "github.com/redhat-developer/app-services-sdk-go/serviceaccounts/apiv1internal/client"
+)
+
+const (
+	accountName        = "serviceAccount"
+	accountDescription = "fake service account"
+)
+
+func CreateServiceAccountForTests(accessToken string, server mocks.RedhatSSOMock, accountName string, accountDescription string) serviceaccountsclient.ServiceAccountData {
+	c := &rhSSOClient{
+		configuration: &serviceaccountsclient.Configuration{
+			DefaultHeader: map[string]string{
+				"Authorization": fmt.Sprintf("Bearer %s", accessToken),
+				"Content-Type":  "application/json",
+			},
+			UserAgent: "OpenAPI-Generator/1.0.0/go",
+			Debug:     false,
+			Servers: serviceaccountsclient.ServerConfigurations{
+				{
+					URL: fmt.Sprintf("%s/auth/realms/redhat-external", server.BaseURL()),
+				},
+			},
+		},
+	}
+	serviceAccount, _ := c.CreateServiceAccount(accessToken, accountName, accountDescription)
+	return serviceAccount
+}
+
+func Test_rhSSOClient_getConfiguration(t *testing.T) {
+	type fields struct {
+		config        *keycloak.KeycloakConfig
+		realmConfig   *keycloak.KeycloakRealmConfig
+		configuration *serviceaccountsclient.Configuration
+		cache         *cache.Cache
+	}
+	type args struct {
+		accessToken string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   *serviceaccountsclient.Configuration
+	}{
+		{
+			name: "should return the clients configuration",
+			fields: fields{
+				config:        &keycloak.KeycloakConfig{},
+				realmConfig:   &keycloak.KeycloakRealmConfig{},
+				configuration: nil,
+				cache:         cache.New(tokenLifeDuration, cacheCleanupInterval),
+			},
+			args: args{
+				accessToken: "accessToken",
+			},
+			want: &serviceaccountsclient.Configuration{
+				DefaultHeader: map[string]string{
+					"Authorization": fmt.Sprintf("Bearer %s", "accessToken"),
+					"Content-Type":  "application/json",
+				},
+				UserAgent: "OpenAPI-Generator/1.0.0/go",
+				Debug:     false,
+				Servers: serviceaccountsclient.ServerConfigurations{
+					{
+						URL: "",
+					},
+				},
+			},
+		},
+	}
```

Comment:
WDYT about testing a scenario where error is returned?

Reply:
that applies to all functions that could return an error

Reply:
Sure @pawelpaszki yea I can add those as for https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1025#issuecomment-1128718326 I kept running into issues when testing them functions but if you think they can be done ill give them another try, and sorry about he spelling mistakes :)

Reply:
no worries. if you are stuck, I can take a look as well